### DISCUSSION TITLE
Ultrawide Stretch Video Alternative

### DISCUSSION THREAD

**Discussion by lokize** [FreeTubeApp/FreeTube#6200] [General]
Hi, is there any way to Stretch video to ultrawide displays?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
At this time there isnt a way to do this but we are looking into this. Options seems to be limited though. 

Just for clarification what do you mean with Stretch,
- Stretch the video but an uniform stretch?
- or only want the edges of the video stretched to fill your peripheral vision?

**Reply by DomiStyle** [reply]
I assume they meant a crop, not a stretch. A lot of videos are published in a 16:9 format but are actually in 21:9 or similar with a black border.

I use the autocrop script in mpv to achieve this but it certainly would be nice to do it natively in FreeTube: https://github.com/mpv-player/mpv/blob/master/TOOLS/lua/autocrop.lua

There's also the ultrawidify browser plugin: https://github.com/tamius-han/ultrawidify

**Comment by purplekairi1312** [comment]
I would like, too, to be able to stretch videos to my ultrawide display. When saying stretching I mean it covering the whole screen (deformed). I would like also if we could be able to change aspect ratio manually like X Y or similar. Putting this here to not open a new 'issue', hope this can help.

**Reply by sheffieldnikki** [reply]
My latest version now has a optional mode for stretch (fill width), in addition to crop (fill height, removing any black bars).

**Comment by sheffieldnikki** [comment]
Just wrote a little PR to crop/zoom videos that normally display with black bars. It doesn't stretch or deform the video. https://github.com/FreeTubeApp/FreeTube/pull/7671

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
duplicate of https://github.com/FreeTubeApp/FreeTube/issues/4310
